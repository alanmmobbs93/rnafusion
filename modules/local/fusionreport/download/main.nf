process FUSIONREPORT_DOWNLOAD {
    tag 'fusionreport'
    label 'process_medium'

    conda "bioconda::star=2.7.9a"
    container "docker.io/clinicalgenomics/fusion-report:3.1.0"

    input:
    val(username)
    val(passwd)

    output:
    path "*"                , emit: reference
    path "versions.yml"     , emit: versions

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args ?: ''
    """
    fusion_report download --cosmic_usr "$username" --cosmic_passwd "$passwd" $args ./

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fusion_report: \$(fusion_report --version | sed 's/fusion-report //')
    END_VERSIONS
    """

    stub:
    """
    touch cosmic.db
    touch fusiongdb2.db
    touch mitelman.db
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fusion_report: \$(fusion_report --version | sed 's/fusion-report //')
    END_VERSIONS
    """

}
