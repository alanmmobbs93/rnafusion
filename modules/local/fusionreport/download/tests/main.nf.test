nextflow_process {

    name "Test Process FUSIONREPORT_DOWNLOAD"
    script "../main.nf"
    process "FUSIONREPORT_DOWNLOAD"

    test("Download fusionreport databases") {

        when {
            process {
                """
                // This process doesn't have any inputs
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.fusiongdb2,
                    process.out.mitelman,
                    process.out.versions
                    ).match() }
            )
        }
    }

    test("Create stub files") {

        options "-stub"

        when {
            process {
                """
                // This process doesn't have any inputs
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
