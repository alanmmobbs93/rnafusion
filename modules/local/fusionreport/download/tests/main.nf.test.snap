{
    "Download fusionreport databases": {
        "content": [
            [
                "fusiongdb2.db:md5,e1ac123a744e515d3e5f85b8344d526a"
            ],
            [
                "mitelman.db:md5,1363795c97f77c641065ecd9ad0e484a"
            ],
            [
                "versions.yml:md5,fa5f13c563f431912048c1802b5a0c74"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-02T13:09:56.930477745"
    },
    "Create stub files": {
        "content": [
            {
                "0": [
                    "fusiongdb2.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "mitelman.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "cosmic.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [
                    "DB-timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "4": [
                    "fusion_report.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "5": [
                    "versions.yml:md5,fa5f13c563f431912048c1802b5a0c74"
                ],
                "cosmic": [
                    "cosmic.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "fusiongdb2": [
                    "fusiongdb2.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log": [
                    "fusion_report.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "mitelman": [
                    "mitelman.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "timestamp": [
                    "DB-timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,fa5f13c563f431912048c1802b5a0c74"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-02T13:10:12.577952658"
    }
}